lnrpc/invoicesrpc/utils.go

```diff
@@ -150,32 +150,38 @@ func CreateRPCInvoice(invoice *invoices.Invoice,
 		rpcHtlcs = append(rpcHtlcs, &rpcHtlc)
 	}
 
+	var blinded *lnrpc.BlindedPathConfig
+	if len(decoded.BlindedPaymentPaths) > 0 {
+		blinded = &lnrpc.BlindedPathConfig{}
+	}
+
 	rpcInvoice := &lnrpc.Invoice{
-		Memo:            string(invoice.Memo),
-		RHash:           rHash,
-		Value:           int64(satAmt),
-		ValueMsat:       int64(invoice.Terms.Value),
-		CreationDate:    invoice.CreationDate.Unix(),
-		SettleDate:      settleDate,
-		Settled:         isSettled,
-		PaymentRequest:  string(invoice.PaymentRequest),
-		DescriptionHash: descHash,
-		Expiry:          int64(invoice.Terms.Expiry.Seconds()),
-		CltvExpiry:      uint64(invoice.Terms.FinalCltvDelta),
-		FallbackAddr:    fallbackAddr,
-		RouteHints:      routeHints,
-		AddIndex:        invoice.AddIndex,
-		Private:         len(routeHints) > 0,
-		SettleIndex:     invoice.SettleIndex,
-		AmtPaidSat:      int64(satAmtPaid),
-		AmtPaidMsat:     int64(invoice.AmtPaid),
-		AmtPaid:         int64(invoice.AmtPaid),
-		State:           state,
-		Htlcs:           rpcHtlcs,
-		Features:        CreateRPCFeatures(invoice.Terms.Features),
-		IsKeysend:       invoice.IsKeysend(),
-		PaymentAddr:     invoice.Terms.PaymentAddr[:],
-		IsAmp:           invoice.IsAMP(),
+		Memo:              string(invoice.Memo),
+		RHash:             rHash,
+		Value:             int64(satAmt),
+		ValueMsat:         int64(invoice.Terms.Value),
+		CreationDate:      invoice.CreationDate.Unix(),
+		SettleDate:        settleDate,
+		Settled:           isSettled,
+		PaymentRequest:    string(invoice.PaymentRequest),
+		DescriptionHash:   descHash,
+		Expiry:            int64(invoice.Terms.Expiry.Seconds()),
+		CltvExpiry:        uint64(invoice.Terms.FinalCltvDelta),
+		FallbackAddr:      fallbackAddr,
+		RouteHints:        routeHints,
+		AddIndex:          invoice.AddIndex,
+		Private:           len(routeHints) > 0,
+		SettleIndex:       invoice.SettleIndex,
+		AmtPaidSat:        int64(satAmtPaid),
+		AmtPaidMsat:       int64(invoice.AmtPaid),
+		AmtPaid:           int64(invoice.AmtPaid),
+		State:             state,
+		Htlcs:             rpcHtlcs,
+		Features:          CreateRPCFeatures(invoice.Terms.Features),
+		IsKeysend:         invoice.IsKeysend(),
+		PaymentAddr:       invoice.Terms.PaymentAddr[:],
+		IsAmp:             invoice.IsAMP(),
+		BlindedPathConfig: blinded,
```

Comment:
this is a little awkward. In a perfect world, there would have been a separate `message AddInvoiceRequest` and `message Invoice` so that creation configuration values could go in the request and then the response can contain pure facts about the invoice. 

Other option is that we do just leave the config struct empty here but add an extra `Blinded bool` (I just dont love the 2 sources of truth)

Reply:
To me having the bool feels a bit cleaner and more direct. We already do this with `IsAmp` btw, since the `Features` field is the primary source of truth, but `IsAmp` makes it easier for the user.

Reply:
added :) thanks for the suggestion to grab this from the feature bits 🙏